### PR TITLE
Add support for float16 buffer in python extension

### PR DESCRIPTION
**Pull request by stevesuzuki-arm**
This PR allows you to pass `numpy.float16` buffer to Halide function.

### CODE REVIEW

**Review by steven-johnson — APPROVED** [review]
LGTM pending green, thanks for the fix!

(I'd love to hear more about what you may be doing with the Python bindings, BTW)

File: src/PythonExtensionGen.cpp [hunk 1]
```diff
@@ -77,6 +77,8 @@ std::pair<string, string> print_type(const LoweredArgument *arg) {
         return std::make_pair("f", "float");
     } else if (arg->type.is_float() && arg->type.bits() == 64) {
         return std::make_pair("d", "double");
+        // } else if (arg->type.is_float() && arg->type.bits() == 16) {
```
[the remark this answers is not in the document]
Comment (stevesuzuki-arm) [reply src/PythonExtensionGen.cpp:80]: Actually, clang-format changed it in this way. Should I add `// clang-format off` here?

[the remark this answers is not in the document]
Comment (steven-johnson) [reply src/PythonExtensionGen.cpp:80]: Ugh. In that case, no, leave it as-is.

**Comment by steven-johnson** [comment]
> > (I'd love to hear more about what you may be doing with the Python bindings, BTW)
> 
> In our use case, Python extension is used in test code because in some case, reference algorithm is in Python and also there are convenient utilities to handle datasets.

**Pull request merged**